Formula/libetpan.rb

```diff
@@ -16,27 +17,91 @@ class Libetpan < Formula
     sha256 cellar: :any, high_sierra:    "6a2f29f42a39d9d3eee7bca1974118fdd8d44a745f61af686aa40c449157b733"
   end
 
-  depends_on xcode: :build
+  depends_on "autoconf@2.69" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "pkg-config" => :build
+
+  uses_from_macos "curl"
+  uses_from_macos "cyrus-sasl"
+  uses_from_macos "expat"
+  uses_from_macos "zlib"
+
+  on_macos do
+    depends_on xcode: :build
+  end
+
+  on_linux do
+    depends_on "openssl@1.1"
+  end
+
+  # Fix extra args in pkg-config file.
+  # Remove with the next release.
+  patch do
+    url "https://github.com/dinhvh/libetpan/commit/8e904aa1c92bd0993123dd46d5a10a58f0516721.patch?full_index=1"
+    sha256 "d99cb79037e147bb376effb6ae5cb1eb46201b0dda6397eba49a1beffc22d9b0"
+  end
+
+  # Fix CVE-2020-15953.
+  # Remove with the next release.
+  patch do
+    url "https://github.com/dinhvh/libetpan/commit/1002a0121a8f5a9aee25357769807f2c519fa50b.patch?full_index=1"
+    sha256 "824408a4d4b59b8e395260908b230232d4f764645b014fbe6e9660ad1137251e"
+  end
+  patch do
+    url "https://github.com/dinhvh/libetpan/commit/298460a2adaabd2f28f417a0f106cb3b68d27df9.patch?full_index=1"
+    sha256 "f5e62879eb90d83d06c4b0caada365a7ea53d4426199a650a7cc303cc0f66751"
+  end
 
   def install
-    xcodebuild "-arch", Hardware::CPU.arch,
-               "-project", "build-mac/libetpan.xcodeproj",
-               "-scheme", "static libetpan",
-               "-configuration", "Release",
-               "SYMROOT=build/libetpan",
-               "build"
-
-    xcodebuild "-arch", Hardware::CPU.arch,
-               "-project", "build-mac/libetpan.xcodeproj",
-               "-scheme", "libetpan",
-               "-configuration", "Release",
-               "SYMROOT=build/libetpan",
-               "build"
-
-    lib.install "build-mac/build/libetpan/Release/libetpan.a"
-    frameworks.install "build-mac/build/libetpan/Release/libetpan.framework"
-    include.install buildpath.glob("build-mac/build/libetpan/Release/include/**")
-    bin.install "libetpan-config"
+    ENV["NOCONFIGURE"] = "1"
+    configure_args = std_configure_args + %w[
+      --without-gnutls
+      --disable-db
+    ]
+
+    # libetpan has a macOS-specific build script to create a framework.
+    # For backwards compatibility reasons, let's continue to do that.
+    if OS.mac?
+      configure_args << "--without-openssl"
+      inreplace "build-mac/update.sh", %r{\./configure (.*) >}, "./configure #{configure_args.join(" ")} >"
+
+      # Fix pkgconfig references pointing to the static library.
+      inreplace "libetpan.pc.in" do |s|
+        s.gsub! "libdir=@libdir@", "libdir=${prefix}/Frameworks"
+        s.gsub! "-L${libdir}", "-F${libdir}"
+        s.gsub! "-letpan@LIBSUFFIX@", "-framework libetpan"
+      end
+
+      cd "build-mac" do
+        # Update the anciently generated autoconf result files.
+        system "./update.sh", "prepare"
+      end
```

Comment:
```suggestion
      # Update the anciently generated autoconf result files.
      system "build-mac/update.sh", "prepare"
```
Can we do this? Or is `update.sh` picky about the working directory?

Reply:
It is very picky.